src/node.c

```diff
@@ -30,9 +30,6 @@ static bool S_can_contain(cmark_node *node, cmark_node *child) {
   if (node == NULL || child == NULL) {
     return false;
   }
-  if (NODE_MEM(node) != NODE_MEM(child)) {
```

Comment:
The same-allocator check for AST nodes is a requirement for a potential memory pool implementation for the nodes. Is it being problematic for this specific PR? If so, feel free to remove that particular check. It can possibly be worked around in the cleanup code for the pool, or I can add it back in the future without much hassle.